### PR TITLE
introduce plugin codegen

### PR DESCRIPTION
**Pull request by joey-colon**
run via `npm run generate plugin "plugin name"` 

e.g.`npm run generate plugin "uc florida"`

### CODE REVIEW

File: scripts/generate.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  return tokens.join('\n').replace(
+    `};`,
+
+    `  ${pluginName.toLowerCase().substr(0, pluginName.length - 6)}: ${pluginName},
+};`
+  );
+};
+
+const createPlugin = async (name) => {
+  const sanitizedName = name.toLowerCase().replace(/ /g, '_');
```
Comment (c650) [inline scripts/generate.js:49]: use `\s`

File: scripts/generate.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    0,
+    `import { ${pluginName} } from '../app/plugins/${fileName}.plugin';`
+  );
+  return tokens.join('\n').replace(
+    `};`,
+
+    `  ${pluginName.toLowerCase().substr(0, pluginName.length - 6)}: ${pluginName},
+};`
+  );
+};
```
Comment (c650) [inline scripts/generate.js:46]: bit sketchy

maybe we should put a comment in `plugin.loader.ts` that says 'do not edit; file managed by generate.js'

**Review by c650 — APPROVED** [review]

File: scripts/generate.js (hunk 2 above)
Comment (joey-colon) [reply scripts/generate.js:46]: we can create a parser to confirm that `plugin.loader.ts` is in the correct format, have it test on circleci, and break build if it isn't.

Comment (c650) [reply scripts/generate.js:46]: agree

**Pull request merged**